doc/source/numpy_2_0_migration_guide.rst

```diff
@@ -83,6 +83,49 @@ used to explicitly implement different behavior on NumPy 1.x and 2.0.
 
 Please let us know if you require additional workarounds here.
 
+.. _migration_c_descr:
+
+The ``PyArray_Descr`` struct has been changed
+---------------------------------------------
+One of the most impactful C-API changes is that the ``PyArray_Descr`` struct
+is now more opaque to allow us to add additional flags and have
+itemsizes not limited by the size of ``int`` as well as allow improving
+structured dtypes in the future and not burdon new dtypes with their fields.
+
+Code which only uses the type number and other initial fields is unaffected.
+Most code will hopefull mainly access the ``->elsize`` field, when the
+dtype/descriptor itself is attached to an array (e.g. ``arr->descr->elsize``)
+this is best replaced with ``PyArray_ITEMSIZE(arr)``.
+
+Where not possible, new accessor functions are required:
+* ``PyDataType_ELSIZE`` and ``PyDataType_SET_ELSIZE`` (note that the result
+  is now ``npy_intp`` and not ``int``).
```

Comment:
Formatting is off in the [rendered page](https://output.circle-artifacts.com/output/job/eff94025-1729-4f27-aca0-11e307460ad7/artifacts/0/doc/build/html/numpy_2_0_migration_guide.html#the-pyarray-descr-struct-has-been-changed)
```suggestion
   is now ``npy_intp`` and not ``int``).
```